core/cog-host-map-handler.c

```diff
@@ -0,0 +1,299 @@
+     * Optional #CogRequestHandler used as fallback when no map entries match
+     * the URI path.
+     */
+    s_properties[PROP_FALLBACK_HANDLER] =
+        g_param_spec_object("fallback-handler",
+                            "Fallback handler",
+                            "Handler used as fallback for unhandled map entries",
+                            COG_TYPE_REQUEST_HANDLER,
+                            G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+
+    g_object_class_install_properties(object_class, N_PROPERTIES, s_properties);
+}
+
+static void
+cog_host_map_handler_init(CogHostMapHandler *self)
+{
+    self->map = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, g_object_unref);
+}
+
+/**
+ * cog_host_map_handler_new: (constructor)
+ * @fallback_handler: (nullable) (transfer none): a #CogRequestHandler
+ *    to use as fallback
+ *
+ * Create a new handler.
+ *
+ * Returns: (transfer full): A handler with no map entries configured.
+ */
+CogRequestHandler *
+cog_host_map_handler_new(CogRequestHandler *fallback_handler)
+{
+    return g_object_new(COG_TYPE_HOST_MAP_HANDLER, "fallback-handler", fallback_handler, NULL);
+}
+
+/**
+ * cog_host_map_handler_contains:
+ * @self: a #CogHostMapHandler
+ * @host: URI host to match
+ *
+ * Check if there is already mounted a route for the @host.
+ *
+ * Returns: Whether the route already exists.
+ */
+gboolean
+cog_host_map_handler_contains(CogHostMapHandler *self, const char *host)
+{
+    g_return_val_if_fail(COG_IS_HOST_MAP_HANDLER(self), FALSE);
+    g_return_val_if_fail(host != NULL, FALSE);
+
+    if (g_hash_table_contains(self->map, host))
+        return TRUE;
+
+    return FALSE;
+}
+
+/**
+ * cog_host_map_handler_mount:
+ * @self: a #CogHostMapHandler
+ * @host: URI host to match
+ * @handler: (transfer none): a request handler for the matched host
+ *
+ * Adds a route to the handler.
+ *
+ * Configures a route which matches @host in URI, and dispatches
+ * requests to a given @handler.
+ *
+ * Returns: Whether the route was successfully added.
+ */
+gboolean
+cog_host_map_handler_mount(CogHostMapHandler *self, const char *host, CogRequestHandler *handler)
+{
+    g_return_val_if_fail(COG_IS_HOST_MAP_HANDLER(self), FALSE);
+    g_return_val_if_fail(host != NULL, FALSE);
+    g_return_val_if_fail(COG_IS_REQUEST_HANDLER(handler), FALSE);
+
+    if (g_hash_table_contains(self->map, host))
+        return FALSE;
+
+    g_hash_table_insert(self->map, g_strdup(host), g_object_ref(handler));
+    return TRUE;
+}
+
+/**
+ * cog_host_map_handler_unmount:
+ * @self: a #CogHostMapHandler
+ * @host: URIhost for a configured route
+ *
+ * Removes a previously configured route.
+ *
+ * Removes a route that matches @host previously configured
+ * using cog_host_map_handler_mount() or
+ * cog_host_map_handler_mount_path().
+ *
+ * Returns: Whether the route was found and removed.
+ */
+gboolean
+cog_host_map_handler_unmount(CogHostMapHandler *self, const char *host)
```

Comment:
Same here, instead of using the “unmount” verb I think it would be better to call this `cog_host_map_handler_remove()`.